src/script/common/c_content.cpp

```diff
@@ -294,6 +294,38 @@ const std::array<const char *, 33> object_property_keys = {
 	"show_on_minimap"
 };
 
+static bool read_engine_mask(lua_State *L, int index, u16 &engine_mask)
```

Comment:
Inserting `read_engine_mask` and `push_engine_mask` in-between `object_property_keys` and `read_object_properties` isn't very sensible. I'd rather add the new functions after `push_object_properties`.